Feat: member 엔티티 구현

vectorch9 opened the pull request:
- [x] Member class 구현
- [x] Gender enum 구현

> table명을 복수로하는 컨벤션과 단수로 하는 컨벤션이 둘다 있어서, 우선 단수로 해놨습니다.
> 이 부분은 상의 후 수정 하겠습니다.

> 추가로 column annotation을 모든 변수에 추가할지도 의견을 주셨으면 합니다.

resolves: #1

vectorch9 committed:
Feat: member 엔티티 구현

Member class, Gender enum 구현

seonpilKim reviewed (commented):
### 테이블명, 컬럼명
- 테이블에는 여러 데이터가 존재하므로 `테이블명은 복수형`으로 사용하는 게 직관적이라 생각합니다.
- 또한, 컬럼명은 `테이블명(단수)_속성명`으로 설계를 해두었으므로, `@Column`으로 컬럼명 지정까지 추가해주시면 좋을 것 같습니다.
    > 참고) 백엔드 1차 회의: https://lilac-freeze-4ae.notion.site/1-9-15-73e0f878d8624b8880ca91085996fc86
### `@Builder` 사용 범위
- 엔티티 클래스에 `@Setter`를 사용하지 않음으로써 얻을 수 있는 효과
   - 중간에 어느 개발자가 임의의 데이터를 변경하지 못하도록 `강제`하는 효과
   - 사용할 수 있는 여지가 있으면, 다른 개발자 입장에서는 사용해도 된다는 것으로 인식할 수 있는 위험성 존재
- 위와 같은 예시를 들었는데, 마찬가지로, `@Builder`를 클래스 단위에 추가하시면, 엔티티 생성 시, `id` 또한 주입이 가능해지기 때문에, 개발자 입장에서는 혼란이 생길 수 있다고 생각합니다.
    - 따라서, 엔티티 생성 시점에 직접 주입할 속성만 골라서 생성자를 하나 만들고, 그 생성자에 `@Builder`를 추가하는 방식이 어떨까 싶습니다
    - `@AllArgsConstructor`도 위와 같은 이유로 제거하면 좋을 것 같습니다.

vectorch9 committed:
Feat: Member 엔티티 개선

userid가 누락되어있어서 추가
ImageType을 Enum을 만들어서 관리
Column 명을 명시하도록 변경
생성자를 AllArgsConstructor에서 필요한 정보만 받도록 수정

vectorch9 commented:
- [x] Member 생성자 수정
- [x] column명 수정
- [x] ImageType enum 구현
- [x] 누락된 userid 추가
> _비고_ 현재 ImageType내에 PNG, JPG, JPEG만 작성했는데, 필요시 추가할 예정

> 지적해주신 사항과 추가적으로 보완할 부분에 대해 개선하였고, 기본 프로필 사진도 구글드라이브를 활용해 업로드 해놓았습니다.
